engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceModificationProcessor.java

```diff
@@ -117,6 +129,27 @@ public void processRecord(
         eventKey, ProcessInstanceModificationIntent.MODIFIED, value, command);
   }
 
+  private Optional<Rejection> validateCommand(
+      final TypedRecord<ProcessInstanceModificationRecord> command, final DeployedProcess process) {
+    final var value = command.getValue();
+
+    final Set<String> unknownElementIds =
+        value.getActivateInstructions().stream()
+            .map(ProcessInstanceModificationActivateInstructionValue::getElementId)
+            .filter(targetElementId -> process.getProcess().getElementById(targetElementId) == null)
+            .collect(Collectors.toSet());
+    if (!unknownElementIds.isEmpty()) {
+      final String reason =
+          String.format(
+              ERROR_MESSAGE_TARGET_ELEMENT_NOT_FOUND,
+              BufferUtil.bufferAsString(process.getBpmnProcessId()),
+              String.join("', '", unknownElementIds));
+      return Optional.of(new Rejection(RejectionType.NOT_FOUND, reason));
```

Comment:
:wrench: We could align the rejection type and use `INVALID_ARGUMENT` instead.

In [CreateProcessInstanceProcessor](https://github.com/camunda/zeebe/blob/ac67c70ed765ad4a010c569516fce656916dd061/engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/CreateProcessInstanceProcessor.java#L181-L197), we reject the command with the type `INVALID_ARGUMENT` if one element id of an activation instruction is unknown. We use `NOT_FOUND` only if the target entity (i.e. the process instance in this case) doesn't exist.

Reply:
That's a good idea. Thanks 👍